tools/rp_storage_tool/src/bucket_reader.rs

```diff
@@ -381,18 +381,29 @@ impl Anomalies {
     }
 
     pub fn merge(&mut self, other: Anomalies) {
-        self.segments_outside_manifest.extend(other.segments_outside_manifest.into_iter());
-        self.archive_manifests_outside_manifest.extend(other.archive_manifests_outside_manifest.into_iter());
-        self.malformed_manifests.extend(other.malformed_manifests.into_iter());
-        self.malformed_lifecycle_markers.extend(other.malformed_lifecycle_markers.into_iter());
-        self.malformed_topic_manifests.extend(other.malformed_topic_manifests.into_iter());
-        self.ntpr_no_manifest.extend(other.ntpr_no_manifest.into_iter());
-        self.ntr_no_topic_manifest.extend(other.ntr_no_topic_manifest.into_iter());
+        self.segments_outside_manifest
+            .extend(other.segments_outside_manifest.into_iter());
+        self.archive_manifests_outside_manifest
+            .extend(other.archive_manifests_outside_manifest.into_iter());
+        self.malformed_manifests
+            .extend(other.malformed_manifests.into_iter());
+        self.malformed_lifecycle_markers
+            .extend(other.malformed_lifecycle_markers.into_iter());
+        self.malformed_topic_manifests
+            .extend(other.malformed_topic_manifests.into_iter());
+        self.ntpr_no_manifest
+            .extend(other.ntpr_no_manifest.into_iter());
+        self.ntr_no_topic_manifest
+            .extend(other.ntr_no_topic_manifest.into_iter());
         self.unknown_keys.extend(other.unknown_keys.into_iter());
-        self.missing_segments.extend(other.missing_segments.into_iter());
-        self.ntpr_bad_deltas.extend(other.ntpr_bad_deltas.into_iter());
-        self.ntpr_overlap_offsets.extend(other.ntpr_overlap_offsets.into_iter());
-        self.metadata_offset_gaps.extend(other.metadata_offset_gaps.into_iter());
+        self.missing_segments
+            .extend(other.missing_segments.into_iter());
+        self.ntpr_bad_deltas
+            .extend(other.ntpr_bad_deltas.into_iter());
+        self.ntpr_overlap_offsets
+            .extend(other.ntpr_overlap_offsets.into_iter());
+        self.metadata_offset_gaps
+            .extend(other.metadata_offset_gaps.into_iter());
```

Comment:
nit: probably `.into_iter()` can be removed from these, just passing the hashset should work for `.extend`, eg

```
self.segments_outside_manifest.extend(other.segments_outside_manifest);
```

Reply:
I cherry picked this commit from Evgeny's [PR](https://github.com/redpanda-data/redpanda/pull/11561), but it's a good suggestion.